### DISCUSSION TITLE
infinite_sheds.get_irradiance running slow for trackers?

### DISCUSSION THREAD

**Discussion by williamhobbs** [williamhobbs/2024_pvpmc_self_shade#11] [Q&A] [answered]
I've noticed that infinite_sheds.get_irradiance() is running really slow for tracking facilities. About 20-50x slower. Is this expected? Or is there something I'm doing wrong?

![image](https://github.com/williamhobbs/2024_pvpmc_self_shade/assets/45701090/5c310143-9ad1-465d-930c-1d11f3ff27c5)

**Comment by kandersolar** [comment, marked as answer]
Yes, expected.  The ground-to-sky view factor inside infinite_sheds is computed in a computationally expensive way.  For FT that cost is paid only once per simulation, but for SAT it is computed for each timestamp, since the tracker rotations change and thus the ground->sky view factor does too.  It's somewhat faster to specify `vectorize=True` (see https://github.com/pvlib/pvlib-python/pull/1682) in the call to infinite_sheds, but it only helps a bit, and uses a lot more memory.  

I know how to fix it (see https://github.com/pvlib/pvlib-python/pull/1865) but it's a pretty big deviation from the reference, and infinite_sheds is already kind of in no-man's-land regarding abiding to a reference, so I've been putting off finishing that PR until we can find time to write a paper for it too.  Anyway yeah it's slow for now :P

**Reply by williamhobbs** [reply]
Thanks. I wondered if it was something like that, but I couldn't see it easily when looking at the infinite sheds source. 

I'm not too worried, mostly wanted to make sure I wasn't doing something wrong.
